docs/src/main/asciidoc/security-oidc-protect-web-application-tutorial.adoc

```diff
@@ -0,0 +1,264 @@
+
+=== Run the application in dev and JVM modes
+
+To run the application in a dev mode, use:
+
+include::{includes}/devtools/dev.adoc[]
+
+When you're done playing with dev mode, you can run it as a standard Java application.
+
+First, compile it:
+
+include::{includes}/devtools/build.adoc[]
+
+Then, run it:
+
+[source,bash]
+----
+java -jar target/quarkus-app/quarkus-run.jar
+----
+
+=== Run the application in Native mode
+
+This same demo can be compiled into native code. 
+No modifications are required.
+
+This implies that you no longer need to install a JVM on your production environment, as the runtime technology is included in
+the produced binary, and optimized to run with minimal resource overhead.
+
+Compilation will take a bit longer, so this step is disabled by default.
+You can build again by enabling the native build:
+
+include::{includes}/devtools/build-native.adoc[]
+
+After getting a cup of coffee, you can run this binary directly:
+
+[source,bash]
+----
+./target/security-openid-connect-web-authentication-quickstart-runner
+----
+
+=== Test the application
+
+To test the application, open your browser and access the following URL:
+
+
+* http://localhost:8080/tokens[http://localhost:8080/tokens]
+
+If everything is working as expected, you are redirected to the Keycloak server to authenticate.
+
+To authenticate to the application, type the following credentials when at the Keycloak login page:
+
+* Username: *alice*
+* Password: *alice*
+
+After clicking the `Login` button, you are redirected back to the application.
+
+// Sheila, JAN 23: TO DO - Update link here to point to new concept topic:For more information about writing the integration tests that depend on `Dev Services for Keycloak`, see the <<integration-testing-keycloak-devservices, Dev Services for Keycloak>> section.
+
+== Summary
+
+Congratulations!
+You have learned how to set up and use the OIDC authorization code flow mechanism to protect and test application HTTP endpoints.
+After you have completed this tutorial, explore some of the other security mechanisms in Quarkus.
+
+* xref:security-oidc-bearer-authentication-concept.adoc[OIDC Bearer authentication]
+* xref:security-overview-concept.adoc[Quarkus Security]
+
+
+== References
+//TO DO, Sheila: update this link when other concept PR is merged - update* xref:security-oidc-web-authentication-concept.adoc[OIDC authorization code flow mechanism]
+* https://www.keycloak.org/documentation.html[Keycloak Documentation]
+* https://openid.net/connect/[OpenID Connect]
+* https://tools.ietf.org/html/rfc7519[JSON Web Token]
+* xref:security-openid-connect-client-reference.adoc[OpenID Connect and OAuth2 Client and Filters Reference Guide]
+* xref:security-openid-connect-dev-services.adoc[Dev Services for Keycloak]
+* xref:security-jwt-build.adoc[Sign and encrypt JWT tokens with SmallRye JWT Build]
+* xref:security-overview-concept.adoc#oidc-jwt-oauth2-comparison[Summary of Quarkus OIDC, JWT and OAuth2 features]
```

Comment:
This target page also got recomposed earlier today too and got caught in the security doc PR crossfire!:fire_extinguisher: 

```suggestion
* xref:security-authentication-mechanisms-concept.adoc#oidc-jwt-oauth2-comparison[Choosing between OpenID Connect, SmallRye JWT, and OAuth2 authentication mechanisms]
```